airlock/business_logic.py

```diff
@@ -1569,6 +1569,16 @@ def check_status(
                 f"cannot change status from {release_request.status.name} to {to_status.name}"
             )
 
+        if to_status == RequestStatus.SUBMITTED:
+            for filegroup in release_request.filegroups:
+                if (
+                    release_request.filegroups[filegroup].context == ""
+                    or release_request.filegroups[filegroup].controls == ""
+                ):
+                    raise self.RequestPermissionDenied(
```

Comment:
used `RequestPermissionDenied` here, because I think that's an appropriate response from `check_status()`, however that means we need to do string matching to catch it in `views/request.py` -> should I instead create a new Exception for this?

Reply:
Yeah, I think the easiest thing would be to make a subclass of `RequestPermissionDenied`.

Another option would be to add an option `reason_code` attribute to the exception and then a new enum type to supply to it. If we start adding lots of different fine grained errors then that would probably be the way to do it. But I think it's too much plumbing when we've only got this one case.

Reply:
> Yeah, I think the easiest thing would be to make a subclass of RequestPermissionDenied

good shout, just done that